features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/ListParameters.jsx

```diff
@@ -64,106 +67,158 @@ export default function ListParameters(props) {
-                                        )}
-                                        <Tooltip title='Delete'>
-                                            <IconButton
-                                                disabled={disableUpdate}
-                                                onClick={() => {
-                                                    operationsDispatcher({
-                                                        action: 'requestBody',
-                                                        data: {
-                                                            target,
-                                                            verb,
-                                                            value: {
-                                                                description: '',
-                                                                required: false,
-                                                                content: {},
-                                                            },
-                                                        },
-                                                    });
-                                                }}
-                                                fontSize='small'
-                                            >
-                                                <DeleteIcon fontSize='small' />
-                                            </IconButton>
-                                        </Tooltip>
-                                    </TableCell>
-                                )}
-                            </TableRow>
-                        ))}
+                                        </TableCell>
+                                    )}
+                                </TableRow>
+                            );
+                        })}
+                    {operation.requestBody && (isRef(operation.requestBody)
+                        ? Object.entries(resolvedSpec.paths[target][verb].requestBody.content).map(
+                            ([contentType, content]) => {
+                                return (
+                                    <RequestBody
+                                        contentType={contentType}
+                                        content={content}
+                                        operation={operation}
+                                        disableUpdate={disableUpdate}
+                                        hideParameterEdit={hideParameterEdit}
+                                        operationsDispatcher={operationsDispatcher}
+                                        target={target}
+                                        verb={verb}
+                                    />
+                                );
+                            },
+                        ) : Object.entries(operation.requestBody.content).map(([contentType, content]) => {
```

Comment:
I think we are repeating the same thing here, Shall we resolve the **requestBody** content in prior to the return statement and pass the content array to **RequestBody** component so that repetitive work like 

```
<RequestBody
                                        contentType={contentType}
                                        content={content}
                                        operation={operation}
                                        disableUpdate={disableUpdate}
                                        hideParameterEdit={hideParameterEdit}
                                        operationsDispatcher={operationsDispatcher}
                                        target={target}
                                        verb={verb}
                                    />
```

and

```
requestBody.content).map(
                            ([contentType, content]) => {
```


can be avoided , 

after getting the **requestBody** content either from a $ref or inline , the content structure is same